gapic/samplegen/samplegen.py

```diff
@@ -551,16 +551,6 @@ def validate_and_transform_request(
                     AttributeRequestSetup(**r_dup)  # type: ignore
                 )
 
-        client_streaming_forms = {
-            types.CallingForm.RequestStreamingClient,
-            types.CallingForm.RequestStreamingBidi,
-        }
-
-        if len(base_param_to_attrs) > 1 and calling_form in client_streaming_forms:
-            raise types.InvalidRequestSetup(
-                "Too many base parameters for client side streaming form"
-            )
-
```

Comment:
@software-dov 👋  I'm having trouble understanding this error, would you remember what this was written to prevent?

Some streaming method request messages have 2+ required fields ([pubsub](https://github.com/googleapis/googleapis/blob/bcfd386c224fded5f5c5be7a17bc5884a2a52f59/google/pubsub/v1/pubsub.proto#L1083-L1162)), so the current snippetgen logic adds two `field`s to the dictionary. That causes this error to be raised. Are fields expected to be set differently for streaming?


```py
# Request passed to `validate_and_transform_request`
[
   {
      "field":"subscription",
      "value":"projects/{project}/subscriptions/{subscription}"
   },
   {
      "field":"stream_ack_deadline_seconds",
      "value":2813
   }
]
```

I see two unit tests that raise this error, but I don't see any pre-existing tests showing a valid sample config for a client streaming method.



https://github.com/googleapis/gapic-generator-python/blob/a8d2b3a2c11984d8677e4e6547a0cbb76d9ed915/tests/unit/samplegen/test_samplegen.py#L1353-L1402

Reply:
I really don't remember what was going on here. My _guess_ is that I was mildly incorrect about the actual python method signature convention and I was trying to imply that we only wanted a single request to be put into a list...?

TL;DR: It's probably a bug or at least a misreading of the spec on my part.